main.py

```diff
@@ -58,19 +67,54 @@ def execute(self):
                                                    connection.protocol.version]
                     version_utils.send_echo(self.controller, switch)
 
+    def _check_overlapping_multipart_request(self, switch):
+        """Check overlapping multipart stats request (OF 1.3 only)."""
+        current_req = self._multipart_replies_xids.get(switch.id, {})
+        if ('flows' in current_req or 'ports' in current_req) and \
+           current_req.get('skipped', 0) < settings.STATS_REQ_SKIP:
+            log.info("Overlapping stats request: switch %s flows_xid %s"
+                     " ports_xid %s", switch.id, current_req.get('flows'),
+                     current_req.get('ports'))
+            current_req['skipped'] = current_req.get('skipped', 0) + 1
+            return True
+
+        if switch.id in self._multipart_replies_flows:
+            del self._multipart_replies_flows[switch.id]
+        if switch.id in self._multipart_replies_ports:
+            del self._multipart_replies_ports[switch.id]
+        return False
+
+    def _get_switch_req_stats_delay(self, switch):
+        if switch.id in self.switch_req_stats_delay:
+            return self.switch_req_stats_delay[switch.id]
+        last = self.switch_req_stats_delay.get('last', 0)
+        max_delay = settings.STATS_INTERVAL/2
+        next_delay = (last + max_delay/10) % max_delay
```

Comment:
The delay distribution func looks good are we planning to stress test this with 30 switches or so (and pushing a significant number of flows), or only when we priorize the queue implementation since that will likely be the holy grail?

Reply:
Thanks for the question, Vinicius. Yes, I would say we better wait to evaluate on a bigger environment when we have features like the queue priority, because there is we will see real benefits. The proposed solution is more of a intermediate step to give us time for a more robust solution.